### ISSUE TITLE
Finetune D0 base on your pre-trained model on coco

### ISSUE THREAD

**Issue by ckqsars** [zylo117/Yet-Another-EfficientDet-Pytorch#211]
I found if I use load the pre-train model and finetune on coco. the map is steady descent.
As I know, if the loss is all right, we finetune a good model. the result is around the pre-trained model.
Here is the result map of finetune model.

the first is finetune 500batch

 Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.306
 Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=100 ] = 0.479
 Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=100 ] = 0.325
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.107
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.355
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.486
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=  1 ] = 0.261
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets= 10 ] = 0.391
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.419
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.165
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.490
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.615

the second is finetune 3500 batch

 Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.169
 Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=100 ] = 0.266
 Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=100 ] = 0.179
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.061
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.193
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.285
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=  1 ] = 0.171
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets= 10 ] = 0.260
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.279
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.103
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.331
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.432

So,is the coco datasets too hard to finetune model which base on your pre-trained model. or other reason?

Any one can tell me why or let me know how to finetune?

**Comment by zylo117** [comment]
It should be around the if you are training it. But these weights are selected. I choosed from many and found the best ones.

**Comment by factplay1** [comment]
> I found if I use load the pre-train model and finetune on coco. the map is steady descent.
> As I know, if the loss is all right, we finetune a good model. the result is around the pre-trained model.
> Here is the result map of finetune model.
> 
> the first is finetune 500batch
> 
> Average Precision (AP) @[ IoU=0.50:0.95 | area= all | maxDets=100 ] = 0.306
> Average Precision (AP) @[ IoU=0.50 | area= all | maxDets=100 ] = 0.479
> Average Precision (AP) @[ IoU=0.75 | area= all | maxDets=100 ] = 0.325
> Average Precision (AP) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.107
> Average Precision (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.355
> Average Precision (AP) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.486
> Average Recall (AR) @[ IoU=0.50:0.95 | area= all | maxDets= 1 ] = 0.261
> Average Recall (AR) @[ IoU=0.50:0.95 | area= all | maxDets= 10 ] = 0.391
> Average Recall (AR) @[ IoU=0.50:0.95 | area= all | maxDets=100 ] = 0.419
> Average Recall (AR) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.165
> Average Recall (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.490
> Average Recall (AR) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.615
> 
> the second is finetune 3500 batch
> 
> Average Precision (AP) @[ IoU=0.50:0.95 | area= all | maxDets=100 ] = 0.169
> Average Precision (AP) @[ IoU=0.50 | area= all | maxDets=100 ] = 0.266
> Average Precision (AP) @[ IoU=0.75 | area= all | maxDets=100 ] = 0.179
> Average Precision (AP) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.061
> Average Precision (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.193
> Average Precision (AP) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.285
> Average Recall (AR) @[ IoU=0.50:0.95 | area= all | maxDets= 1 ] = 0.171
> Average Recall (AR) @[ IoU=0.50:0.95 | area= all | maxDets= 10 ] = 0.260
> Average Recall (AR) @[ IoU=0.50:0.95 | area= all | maxDets=100 ] = 0.279
> Average Recall (AR) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.103
> Average Recall (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.331
> Average Recall (AR) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.432
> 
> So,is the coco datasets too hard to finetune model which base on your pre-trained model. or other reason?
> 
> Any one can tell me why or let me know how to finetune?

That's a good experiment to do.
I also have troubles with training on COCO - so far have tried many set of hyper-parameters.

**Comment by ckqsars** [comment]
> It should be around the if you are training it. But these weights are selected. I choosed from many and found the best ones.

ye, in the first 500 batch, the result is around your best map, but when I finetune to 3500 batch the map is only half of your result.

In a general way, When we finetune a model on the best weights, the finetuned model performance is similar to the best weighted model.

**Comment by zylo117** [comment]
If you are using the parameters that's at its final stage, yes, otherwise no. The parameters in the training script is for custom dataset or training from scratch.

**Comment by iamastar88** [comment]
So, can you share the final stage parameters on coco, thanks.
